CRM/Contribute/BAO/Contribution.php

```diff
@@ -3592,6 +3604,28 @@ public static function updateFinancialAccounts(&$params, $context = NULL, $skipT
         }
       }
     }
+    if ($context == 'changePaymentInstrument') {
+      foreach ($params['line_item'] as $lineitems) {
+        foreach ($lineitems as $fieldValueId => $fieldValues) {
+          $prevFinancialItem = CRM_Financial_BAO_FinancialItem::getPreviousFinancialItem($fieldValues['id']);
+          if (!CRM_Utils_Rule::currencyCode($trxn->currency)) {
+            $trxn->currency = CRM_Core_Config::singleton()->defaultCurrency;
+          }
+
+          // save to entity_financial_trxn table
+          $entityFinancialTrxnParams = array(
+            'entity_table' => "civicrm_financial_item",
+            'entity_id' => $prevFinancialItem->id,
+            'financial_trxn_id' => $trxn->id,
+            'amount' => $trxn->total_amount,
+            'currency' => $trxn->currency,
```

Comment:
this is as per https://github.com/civicrm/civicrm-core/blob/2fa84df330aecce758529af45c2e1986b025e178/CRM/Core/BAO/FinancialTrxn.php#L69